lnd.go

```diff
@@ -1135,3 +1115,93 @@ func waitForWalletPassword(cfg *Config, restEndpoints []net.Addr,
 		return nil, fmt.Errorf("shutting down")
 	}
 }
+
+// initializeDatabases extracts the current databases that we'll use for normal
+// operation in the daemon. Two databases are returned: one remote and one
+// local. However, only if the replicated database is active will the remote
+// database point to a unique database. Otherwise, the local and remote DB will
```

Comment:
With the change to move the height hints to the local DB, `GraphDB` doesn't _quite_ work here. I agree in general though, as then the "functional" splits let us start to abstract away those portions, so being able to manipulate the graph on an interface level which'll let us use more specific DB features rather than mapping everything unto the `kvdb` interface.  In any case I don't think this is blocking, since we're not exposing any new config options, they're (for now) just internal variable names.